src/libstore/meson.build

```diff
@@ -0,0 +1,202 @@
+    join_paths(libstore_dir, 'lock.cc'),
+    join_paths(libstore_dir, 'machines.cc'),
+    join_paths(libstore_dir, 'misc.cc'),
+    join_paths(libstore_dir, 'names.cc'),
+    join_paths(libstore_dir, 'nar-accessor.cc'),
+    join_paths(libstore_dir, 'nar-info.cc'),
+    join_paths(libstore_dir, 'nar-info-disk-cache.cc'),
+    join_paths(libstore_dir, 'optimise-store.cc'),
+    join_paths(libstore_dir, 'parsed-derivations.cc'),
+    join_paths(libstore_dir, 'path.cc'),
+    join_paths(libstore_dir, 'pathlocks.cc'),
+    join_paths(libstore_dir, 'profiles.cc'),
+    join_paths(libstore_dir, 'realisation.cc'),
+    join_paths(libstore_dir, 'references.cc'),
+    join_paths(libstore_dir, 'remote-fs-accessor.cc'),
+    join_paths(libstore_dir, 'remote-store.cc'),
+    join_paths(libstore_dir, 's3-binary-cache-store.cc'),
+    join_paths(libstore_dir, 'sqlite.cc'),
+    join_paths(libstore_dir, 'ssh.cc'),
+    join_paths(libstore_dir, 'ssh-store.cc'),
+    join_paths(libstore_dir, 'store-api.cc'),
+    join_paths(libstore_dir, 'uds-remote-store.cc'))
+
+libstore_headers = files(
+    join_paths(libstore_dir, 'binary-cache-store.hh'),
+    join_paths(libstore_dir, 'builtins.hh'),
+    join_paths(libstore_dir, 'crypto.hh'),
+    join_paths(libstore_dir, 'daemon.hh'),
+    join_paths(libstore_dir, 'derivations.hh'),
+    join_paths(libstore_dir, 'filetransfer.hh'),
+    join_paths(libstore_dir, 'fs-accessor.hh'),
+    join_paths(libstore_dir, 'globals.hh'),
+    join_paths(libstore_dir, 'local-store.hh'),
+    join_paths(libstore_dir, 'machines.hh'),
+    join_paths(libstore_dir, 'names.hh'),
+    join_paths(libstore_dir, 'nar-accessor.hh'),
+    join_paths(libstore_dir, 'nar-info-disk-cache.hh'),
+    join_paths(libstore_dir, 'nar-info.hh'),
+    join_paths(libstore_dir, 'parsed-derivations.hh'),
+    join_paths(libstore_dir, 'path.hh'),
+    join_paths(libstore_dir, 'pathlocks.hh'),
+    join_paths(libstore_dir, 'profiles.hh'),
+    join_paths(libstore_dir, 'references.hh'),
+    join_paths(libstore_dir, 'remote-fs-accessor.hh'),
+    join_paths(libstore_dir, 'remote-store.hh'),
+    join_paths(libstore_dir, 's3-binary-cache-store.hh'),
+    join_paths(libstore_dir, 's3.hh'),
+    join_paths(libstore_dir, 'serve-protocol.hh'),
+    join_paths(libstore_dir, 'sqlite.hh'),
+    join_paths(libstore_dir, 'ssh.hh'),
+    join_paths(libstore_dir, 'store-api.hh'),
+    join_paths(libstore_dir, 'worker-protocol.hh'))
+
+
+libstore_data = files(
+    join_paths(libstore_dir, 'sandbox-defaults.sb'),
+    join_paths(libstore_dir, 'sandbox-minimal.sb'),
+    join_paths(libstore_dir, 'sandbox-network.sb'))
+
+
+# include directories
+#========================================================================
+
+libstore_dirs = [
+    'build',
+    'builtins']
+
+
+foreach dir : libstore_dirs
+    subdir(dir)
+endforeach
+
+
+# targets
+#============================================================================
+
+libstore_src += custom_target(
```

Comment:
Will `meson` recompile `local-store-cc` when those change?
See the current dependencies https://github.com/NixOS/nix/blob/master/src/libstore/local.mk#L51 .

Reply:
im not sure what you mean :)

Reply:
Seems that with the current `local.mk`, `local-store-cc` will be recompiled if any of those `.hh` is updated.
Is that the case with `meson` and its current setup?

Reply:
as far as i know, the files are generated with each build into the build directory. if the files change, it regenerates the build files, and rebuilds any dependencies

Reply:
> and rebuilds any dependencies

My question is: "are those two `.hh` dependencies of `local-store.cc` in the current `meson` setup"?

Reply:
they are not :)